src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalSuggestAddon.ts

```diff
@@ -306,51 +306,71 @@ export class SuggestAddon extends Disposable implements ITerminalAddon, ISuggest
 
 	private _sync(promptInputState: IPromptInputModelState): void {
 		const config = this._configurationService.getValue<ITerminalSuggestConfiguration>(terminalSuggestConfigSection);
-		if (!this._mostRecentPromptInputState || promptInputState.cursorIndex > this._mostRecentPromptInputState.cursorIndex) {
-			// If input has been added
+		{
 			let sent = false;
 
-			// Quick suggestions - Trigger whenever a new non-whitespace character is used
-			if (!this._terminalSuggestWidgetVisibleContextKey.get()) {
-				if (config.quickSuggestions) {
-					if (promptInputState.prefix.match(/[^\s]$/)) {
+			// If the cursor moved to the right
+			if (!this._mostRecentPromptInputState || promptInputState.cursorIndex > this._mostRecentPromptInputState.cursorIndex) {
+				// Quick suggestions - Trigger whenever a new non-whitespace character is used
+				if (!this._terminalSuggestWidgetVisibleContextKey.get()) {
```

Comment:
No changes actually happened in this block, it's just indented since it's in a plain `{}` block so we can keep `sent` scoped to just this part